Test suite leaks container networks during execution

Running `KafkaClusterTest` with env var `TEST_CLUSTER_EXECUTION_MODE=true` whilst watching the container networks (`docker network ls`), I notice that the container networks accumulate.  One per test. 

```
TEST_CLUSTER_EXECUTION_MODE=CONTAINER mvn -Dtest=KafkaClusterTest test -Dsurefire.failIfNoSpecifiedTests=false
```

<img width="1209" alt="Screenshot 2023-07-06 at 13 56 16" src="https://github.com/kroxylicious/kroxylicious-junit5-extension/assets/18440250/9acd2f50-ce47-4ffe-8492-8fcca56988b4">

The networks get torn down when the JVM exits (thanks to hooks installed by Testcontainers, I think).  However, if I add more tests, eventually, Docker fail with a message like the below.

Network used to be a static, it was made instance since (1fb8e7289cea00a1ad44d27f0791abe800087683).


```
Caused by: com.github.dockerjava.api.exception.NotFoundException: Status 404: {"message":"could not find an available, non-overlapping IPv4 address pool among the defaults to assign to the network"}

	at org.testcontainers.shaded.com.github.dockerjava.core.DefaultInvocationBuilder.execute(DefaultInvocationBuilder.java:241)
	at org.testcontainers.shaded.com.github.dockerjava.core.DefaultInvocationBuilder.post(DefaultInvocationBuilder.java:124)
	at org.testcontainers.shaded.com.github.dockerjava.core.exec.CreateNetworkCmdExec.execute(CreateNetworkCmdExec.java:27)
	at org.testcontainers.shaded.com.github.dockerjava.core.exec.CreateNetworkCmdExec.execute(CreateNetworkCmdExec.java:12)
	at org.testcontainers.shaded.com.github.dockerjava.core.exec.AbstrSyncDockerCmdExec.exec(AbstrSyncDockerCmdExec.java:21)
	at org.testcontainers.shaded.com.github.dockerjava.core.command.AbstrDockerCmd.exec(AbstrDockerCmd.java:35)
	at org.testcontainers.containers.Network$NetworkImpl.create(Network.java:100)
	at org.testcontainers.containers.Network$NetworkImpl.getId(Network.java:64)
	at org.testcontainers.containers.GenericContainer.applyConfiguration(GenericContainer.java:879)
	at org.testcontainers.containers.GenericContainer.tryStart(GenericContainer.java:378)
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -6,6 +6,7 @@ Please enumerate all user-facing changes using format `<githib issue/pr number>:
 
 ## 0.5.0
 
+* [#140](https://github.com/kroxylicious/kroxylicious-junit5-extension/issues/140) Prevent docker networks accumulations during test runs.
 * [#134](https://github.com/kroxylicious/kroxylicious-junit5-extension/pull/134): Support @BeforeEach and test cases referring to same cluster
 * [#130](https://github.com/kroxylicious/kroxylicious-junit5-extension/pull/130): Small KeytoolCertificateGenerator improvements (eliminates use of openssl)
 

--- a/impl/src/main/java/io/kroxylicious/testing/kafka/testcontainers/TestcontainersKafkaCluster.java
+++ b/impl/src/main/java/io/kroxylicious/testing/kafka/testcontainers/TestcontainersKafkaCluster.java
@@ -341,7 +341,12 @@ public class TestcontainersKafkaCluster implements Startable, KafkaCluster, Kafk
 
     @Override
     public synchronized void stop() {
-        allContainers().parallel().forEach(GenericContainer::stop);
+        try {
+            allContainers().parallel().forEach(GenericContainer::stop);
+        }
+        finally {
+            network.close();
+        }
     }
 
     @Override

